tests/k8st/tests/test_bgp_advert.py

```diff
@@ -165,7 +165,42 @@ def test_mainline(self):
         retry_until_success(lambda: self.assertIn(local_svc_ip, self.get_routes()))
         retry_until_success(lambda: self.assertNotIn(cluster_svc_ip, self.get_routes()))
 
-        # Delete nginx service
+        # Create a network policy that only accepts traffic from the external node
+        run("""kubectl exec -i -n kube-system calicoctl -- /calicoctl apply -f - << EOF
```

Comment:
This is fine, but it might be easier to use a k8s NP rather than a Calico one - that way, it gets automatically deleted when the namespace is deleted.